tests/sentry/hybrid_cloud/test_replica.py

```diff
@@ -0,0 +1,181 @@
+        )
+
+    assert replicated.auth_identity_id == auth_identity.id
+    assert replicated.data == auth_identity.data
+
+    with assume_test_silo_mode(SiloMode.CONTROL):
+        auth_identities = [
+            auth_identity,
+            AuthIdentity.objects.create(
+                user=user2, auth_provider=auth_provider, ident="some-ident-2", data={"b": 2}
+            ),
+            AuthIdentity.objects.create(
+                user=user3, auth_provider=auth_provider, ident="some-ident-3", data={"b": 2}
+            ),
+        ]
+        auth_idents = [ai.ident for ai in auth_identities]
+        conflicting_pairs = list(zip(auth_identities, [*auth_idents[1:], auth_idents[0]]))
+
+        with outbox_runner(), outbox_context(flush=False):
+            for ai in auth_identities:
+                ai.ident += "-new"
+                ai.save()
+
+            for ai, next_ident in conflicting_pairs:
+                ai.ident = next_ident
+                ai.save()
+
+        with assume_test_silo_mode(SiloMode.REGION):
+            for ai, next_ident in zip(auth_identities, [*auth_idents[1:], auth_idents[0]]):
+                assert AuthIdentityReplica.objects.get(auth_identity_id=ai.id).ident == next_ident
+
+
+@django_db_all(transaction=True)
+@all_silo_test(stable=True)
+def test_replicate_team():
+    org = Factories.create_organization()
+    with assume_test_silo_mode(SiloMode.CONTROL):
+        assert TeamReplica.objects.count() == 0
+
+    with assume_test_silo_mode(SiloMode.REGION):
+        team = Factories.create_team(org)
+
+    with assume_test_silo_mode(SiloMode.CONTROL):
+        replicated = TeamReplica.objects.get(team_id=team.id)
+
+    assert replicated.organization_id == team.organization_id
+    assert replicated.slug == team.slug
+    assert replicated.name == team.name
+    assert replicated.status == team.status
+    assert replicated.org_role == team.org_role
+
+    with assume_test_silo_mode(SiloMode.REGION):
+        team.org_role = "boo"
+        team.save()
+
+    with assume_test_silo_mode(SiloMode.CONTROL):
+        replicated = TeamReplica.objects.get(team_id=team.id)
+
+    assert replicated.org_role == team.org_role
+
+    with assume_test_silo_mode(SiloMode.REGION):
+        teams = [
+            team,
+            Factories.create_team(organization=team.organization),
+            Factories.create_team(organization=team.organization),
+        ]
+        team_slugs = [team.slug for team in teams]
+        conflicting_pairs = list(zip(teams, [*team_slugs[1:], team_slugs[0]]))
+
+        with outbox_runner(), outbox_context(flush=False):
+            for team in teams:
+                team.slug += "-new"
+                team.save()
+
+            for team, next_slug in conflicting_pairs:
+                team.slug = next_slug
+                team.save()
+
+        with assume_test_silo_mode(SiloMode.CONTROL):
+            for team, next_slug in zip(teams, [*team_slugs[1:], team_slugs[0]]):
```

Comment:
What about the scenario where a single team is renamed? For example alpha -> bravo. The receiver looks like it would end up with both alpha and bravo in the teamreplica as the receiver removes bravo, and then reinserts bravo again, but doesn't remove `alpha`.

Reply:
This also makes me think I probably CAN generalize the replication a bit to guard these cases correctly by default.  But that's for a future PR.